### PR TITLE
Remove unused state form TypeofBinder and GetTypeBinder.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler @dotnet/roslyn-ide ptal.

File: src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs:26]: Unused field. (only read in unused property).

File: src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs:31]: unused out param (only used to set field that is never read).

File: src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs:35]: unread property.

File: src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs:54]: field used only to set out parameter which is never used.

File: src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharpTest/CodeGeneration/SyntaxGeneratorTests.cs:42]: ide tests.

File: src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs
Comment (CyrusNajmabadi) [reply src/Compilers/CSharp/Portable/Binder/TypeofBinder.cs:26]: discovered while debugging through compiler to determine how unbound generics are bound.

**Comment by 333fred** [comment]
Could you please update the description of this PR? Is it fixing a bug?

**Comment by CyrusNajmabadi** [comment]
> Could you please update the description of this PR?

It simplifies creation of these types in the compiler, by not having it do unnecessary work, or store unused/unread state.

This was helpful for me when trying to debug an issue reported by a customer related to unbound generics when i was going through the compiler.  I was trying to navigate through how all these values worked and what purpose they served.  Discovering that these were not used (After having to have them in my mental map of this process) def occupied extra cycles for me.  The goal here was to remove that to help maintainers and future people trying to understand and work with this code.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 1), modulo:
- It doesn't look like the title "Simplify creation of unbound generic types" accurately reflects the nature of the change. Consider using more meaningful name, "Remove unused state form TypeofBinder and GetTypeBinder.", for example.
- It looks like the change in `SyntaxGeneratorTests` doesn't depend on the cleanup/refactoring in compilers and, therefore, is unrelated. Consider separating it into a dedicated PR.

**Comment by CyrusNajmabadi** [comment]
> Remove unused state form TypeofBinder and GetTypeBinder.

Done.

**Comment by CyrusNajmabadi** [comment]
> It looks like the change in SyntaxGeneratorTests doesn't depend on the cleanup/refactoring in compilers and, therefore, is unrelated. Consider separating it into a dedicated PR.

Sure.

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 3)

**Review by 333fred — APPROVED** [review]

**Pull request merged**
